e2e_tests/test_suites/inventory/inventory.go

```diff
@@ -278,6 +278,13 @@ func inventoryTestCase(ctx context.Context, testSetup *inventoryTestSetup, tests
 	ga, ok := runGatherInventoryTest(ctx, testSetup, gatherInventoryTest, &logwg)
 	gatherInventoryTest.Finish(tests)
 	logger.Printf("TestCase %q finished", gatherInventoryTest.Name)
+	if !ok {
+		rerunTC := junitxml.NewTestCase(testSuiteName, strings.TrimPrefix(gatherInventoryTest.Name, fmt.Sprintf("[%s] ", testSuiteName)))
+		logger.Printf("Rerunning TestCase %q", rerunTC.Name)
+		ga, ok = runGatherInventoryTest(ctx, testSetup, rerunTC, &logwg)
+		rerunTC.Finish(tests)
+		logger.Printf("TestCase %q finished in %fs", rerunTC.Name, rerunTC.Time)
+	}
```

Comment:
It's a little confusing to have multiple `if !ok {` blocks in a row. Should the next one be nested after line 284?